.github/workflows/steel-documentation.yml

```
@@ -0,0 +1,55 @@
+name: Deploy Steel Documentation
+
+on:
+  push:
+    branches: ["main"]
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow only one concurrent deployment
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
+
+env:
+  RISC0_TOOLCHAIN_VERSION: r0.1.81.0
+  RISC0_MONOREPO_REF: "main"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Setup Pages
+        uses: actions/configure-pages@v4
+      
+      - uses: risc0/risc0/.github/actions/rustup@main    
+
+      - name: Build documentation
+        run: cargo doc --no-deps --document-private-items
```

I don't think we want to show private items here. They can read the source code if they'd like that level of detail.
```suggestion
        run: cargo doc --no-deps
```